src/librustc/hir/map/mod.rs

```diff
@@ -347,10 +347,19 @@ impl<'hir> Map<'hir> {
                 };
                 DefKind::Ctor(ctor_of, def::CtorKind::from_hir(variant_data))
             }
-            Node::AnonConst(_)
-            | Node::Field(_)
-            | Node::Expr(_)
-            | Node::Stmt(_)
+            Node::AnonConst(_) => DefKind::AnonConst,
+            Node::Field(_) => DefKind::Field,
+            Node::Expr(expr) => match expr.kind {
+                ExprKind::Closure { .. } => DefKind::Closure,
+                _ => return None,
```

Comment:
If you make this method take `LocalDefId` then you can `bug!` here, because this case would be impossible.